File: src/main/java/exnihilocreatio/json/CustomEntityInfoJson.java
package exnihilocreatio.json;

import com.google.gson.*;
import exnihilocreatio.util.EntityInfo;

import java.lang.reflect.Type;

public class CustomEntityInfoJson implements JsonDeserializer<EntityInfo>, JsonSerializer<EntityInfo> {
    @Override
    public JsonElement serialize(EntityInfo src, Type typeOfSrc, JsonSerializationContext context) {
        return new JsonPrimitive(src.getName());
    }

    @Override
    public EntityInfo deserialize(JsonElement json, Type typeOfT, JsonDeserializationContext context) throws JsonParseException {
        String name = json.getAsString();
        return new EntityInfo(name);
    }
}

File: src/main/java/exnihilocreatio/json/CustomEntityInfoJson.kt
package exnihilocreatio.json

import com.google.gson.*
import exnihilocreatio.util.EntityInfo

import java.lang.reflect.Type

object CustomEntityInfoJson : JsonDeserializer<EntityInfo>, JsonSerializer<EntityInfo> {
    override fun serialize(src: EntityInfo, typeOfSrc: Type, context: JsonSerializationContext): JsonElement =
            JsonPrimitive(src.name)

    @Throws(JsonParseException::class)
    override fun deserialize(json: JsonElement, typeOfT: Type, context: JsonDeserializationContext): EntityInfo =
            EntityInfo(json.asString)

}
